app/src/main/java/org/schabi/newpipe/ui/Toolbar.kt

```diff
@@ -53,60 +68,95 @@ fun SearchSuggestionItem(text: String) {
 fun Toolbar(
     title: String,
     modifier: Modifier = Modifier,
-    hasNavigationIcon: Boolean = true,
+    onNavigateBack: (() -> Unit)? = null,
     hasSearch: Boolean = false,
-    onSearchQueryChange: ((String) -> List<String>)? = null,
+    onSearch: (String) -> Unit,
+    searchResults: List<String>,
     actions: @Composable RowScope.() -> Unit = {}
 ) {
     var isSearchActive by remember { mutableStateOf(false) }
-    var query by remember { mutableStateOf("") }
+    var expanded by rememberSaveable { mutableStateOf(false) }
+    val textFieldState = rememberTextFieldState()
 
     Column {
         TopAppBar(
             title = { Text(text = title) },
             modifier = modifier,
-            navigationIcon = { if (hasNavigationIcon) NavigationIcon() },
+            colors = TopAppBarDefaults.topAppBarColors(
+                containerColor = MaterialTheme.colorScheme.primary,
+                titleContentColor = MaterialTheme.colorScheme.onPrimary,
+            ),
```

Comment:
Better remove this, since I have raised a PR to migrate to the Material 3 color scheme: https://github.com/TeamNewPipe/NewPipe/pull/12654